pkg/auth/requests/impersonate_test.go

```diff
@@ -0,0 +1,291 @@
+				req := &http.Request{
+					Header: map[string][]string{
+						"Impersonate-User":      {"impUser"},
+						"Impersonate-Extra-foo": {"bar"},
+					},
+				}
+				req = req.WithContext(ctx)
+
+				return req
+			},
+			sarMock: func(req *http.Request) sar.SubjectAccessReview {
+				mock := mocks.NewMockSubjectAccessReview(ctrl)
+				mock.EXPECT().UserCanImpersonateUser(req, "user", "impUser").Return(true, nil)
+				mock.EXPECT().UserCanImpersonateExtras(req, "user", map[string][]string{"foo": {"bar"}}).Return(false, nil)
+
+				return mock
+			},
+			expectedInfo:   nil,
+			expectedAuthed: false,
+			expectedErr:    "not allowed to impersonate extra",
+		},
+		"impersonate user error": {
+			req: func() *http.Request {
+				ctx := request.WithUser(context.Background(), userInfo)
+				req := &http.Request{
+					Header: map[string][]string{
+						"Impersonate-User": {"impUser"},
+					},
+				}
+				req = req.WithContext(ctx)
+
+				return req
+			},
+			sarMock: func(req *http.Request) sar.SubjectAccessReview {
+				mock := mocks.NewMockSubjectAccessReview(ctrl)
+				mock.EXPECT().UserCanImpersonateUser(req, "user", "impUser").Return(false, errors.New("unexpected error"))
+
+				return mock
+			},
+			expectedInfo:   nil,
+			expectedAuthed: false,
+			expectedErr:    "unexpected error",
+		},
+		"impersonate group error": {
+			req: func() *http.Request {
+				ctx := request.WithUser(context.Background(), userInfo)
+				req := &http.Request{
+					Header: map[string][]string{
+						"Impersonate-User":  {"impUser"},
+						"Impersonate-Group": {"impGroup"},
+					},
+				}
+				req = req.WithContext(ctx)
+
+				return req
+			},
+			sarMock: func(req *http.Request) sar.SubjectAccessReview {
+				mock := mocks.NewMockSubjectAccessReview(ctrl)
+				mock.EXPECT().UserCanImpersonateUser(req, "user", "impUser").Return(true, nil)
+				mock.EXPECT().UserCanImpersonateGroup(req, "user", "impGroup").Return(false, errors.New("unexpected error"))
+
+				return mock
+			},
+			expectedInfo:   nil,
+			expectedAuthed: false,
+			expectedErr:    "unexpected error",
+		},
+		"impersonate extras error": {
+			req: func() *http.Request {
+				ctx := request.WithUser(context.Background(), userInfo)
+				req := &http.Request{
+					Header: map[string][]string{
+						"Impersonate-User":      {"impUser"},
+						"Impersonate-Extra-foo": {"bar"},
+					},
+				}
+				req = req.WithContext(ctx)
+
+				return req
+			},
+			sarMock: func(req *http.Request) sar.SubjectAccessReview {
+				mock := mocks.NewMockSubjectAccessReview(ctrl)
+				mock.EXPECT().UserCanImpersonateUser(req, "user", "impUser").Return(true, nil)
+				mock.EXPECT().UserCanImpersonateExtras(req, "user", map[string][]string{"foo": {"bar"}}).Return(false, errors.New("unexpected error"))
+
+				return mock
+			},
+			expectedInfo:   nil,
+			expectedAuthed: false,
+			expectedErr:    "unexpected error",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
```

Comment:
This works correctly with `t.Parrallel()` only since Go 1.22. Depending on the backport needs we may want make a properly scoped copy of the test var
```suggestion
		test := test
		t.Run(name, func(t *testing.T) {
```